qualtran/bloqs/chemistry/sparse/prepare.py

```diff
@@ -340,75 +345,48 @@ def build_qrom_bloq(self) -> 'Bloq':
             self.keep,
             target_bitsizes=target_bitsizes,
             log_block_sizes=log_block_sizes,
-            use_dirty_ancilla=False,
         )
+        # if self.is_adjoint:
+        #     return qrom.adjoint()
         return qrom
 
-    def build_composite_bloq(
-        self,
-        bb: 'BloqBuilder',
-        d: 'SoquetT',
-        p: 'SoquetT',
-        q: 'SoquetT',
-        r: 'SoquetT',
-        s: 'SoquetT',
-        sigma: 'SoquetT',
-        alpha: 'SoquetT',
-        beta: 'SoquetT',
-        rot_aa: 'SoquetT',
-        swap_pq: 'SoquetT',
-        swap_rs: 'SoquetT',
-        swap_pqrs: 'SoquetT',
-        flag_1b: 'SoquetT',
-        alt_pqrs: NDArray[Soquet],  # type: ignore[type-var]
-        theta: NDArray[Soquet],  # type: ignore[type-var]
-        keep: 'SoquetT',
-        less_than: 'SoquetT',
-        alt_flag_1b: 'SoquetT',
-    ) -> Dict[str, 'SoquetT']:
+    def add_qrom(self, bb: 'BloqBuilder', **soqs: 'SoquetT') -> Dict[str, 'SoquetT']:
+        qrom = self.build_qrom_bloq()
+        # The qroam_junk_regs won't be present initially when building the
+        # composite bloq as they're RIGHT registers.
```

Comment:
Ditto: Please update to `qroam_target` and `qroam_junk` correspondingly; instead of `qroam_junk` and `qroam_extra_junk`.